src/lambda_streams/Animals.java

```diff
@@ -26,37 +31,78 @@ public static void main(String[] args) {
+            return copy;
+        }
+
     }
 
     static String addAnimal(String animal) {
         //add an animal to the animal list.
+        List<String> newAnimal = new ArrayList<>(animals);
+
+        newAnimal.add(animal);
+        animals = newAnimal;
         return animal;
     };
 
 
     static List<String> lowerFirst(List<String> animaList, boolean mutate) {
         //lowercase the first letter, and uppercase the rest of the letters, using streams.  Also, depending on the value of the boolean flag 'mutate', mutate the original animals list, or perform your stream operations on a 'copy' of the animals list.  return the list out of hte function in both cases.
-        return animaList;
+        if(mutate) {
+            animaList.stream().map(str -> str.substring(0, 1).toLowerCase() + str.substring(1).toUpperCase()).collect(Collectors.toList());
+            return animaList;
+        }
+        else {
+            List<String> copy = animaList.stream().map(str -> str.substring(0, 1).toLowerCase() + str.substring(1).toUpperCase()).collect(Collectors.toList());
+            return copy;
+        }
     }
 
     static List<String> flipAnimals(boolean mutate) {
         //reverse the order of the animals in the animal list.  If the booleaen parameter is true, reverse the static animals array list by mutating the array.  if the mutate boolean is false, flip a 'copy' of the animals list, then return that list of flipped animals, WITHOUT mutating the static animals array. return the flipped list in both cases.
 
         List<String> instructions = Arrays.asList("Flip","the","animals","list","."," ","Mutate","the","animals","only","if","flag","true");
-        return instructions;
+
+        if(mutate) {
+            Collections.reverse(animals);
+            return animals;
+        }
+        else {
+            List<String> newAnimal = new ArrayList<>(animals);
+            Collections.reverse(newAnimal);
+            return newAnimal;
+
+        }
     }
 
     static List<String> sortAnimals(boolean mutate) {
-    //sort the animals in alphabetical order.  If the booleaen parameter is true, mutating the animals list.  if the mutate boolean is false, sort a 'copy' of the animals list, then return that list of sorted animals, WITHOUT mutating the static animals array. return the sorted list in both cases.
-    return animals;
+        //sort the animals in alphabetical order.  If the booleaen parameter is true, mutating the animals list.  if the mutate boolean is false, sort a 'copy' of the animals list, then return that list of sorted animals, WITHOUT mutating the static animals array. return the sorted list in both cases.
+        if(mutate) {
+            animals.sort(Comparator.naturalOrder());
+            return animals;
+        }
+        else {
+            List<String> newList = new ArrayList<>(animals);
+            newList.sort(Comparator.naturalOrder());
+
+            return newList;
+        }
+
+
     }
 
 
-}
+
```

Comment:
It was expected that all methods be refactored to take a lambda as an additional argument, then use the lambda to return the result.  None of the methods have been refactored.